activerecord/lib/active_record/persistence.rb

```diff
@@ -280,15 +280,26 @@ def instantiate(attributes, column_types = {}, &block)
       #   people.update(group: "masters")
       #
       # Note: Updating a large number of records will run an UPDATE
-      # query for each record, which may cause a performance issue.
+      # query for each record(duplicates will be merged together first),
+      # which may cause a performance issue.
       # When running callbacks is not needed for each record update,
       # it is preferred to use {update_all}[rdoc-ref:Relation#update_all]
       # for updating all records in a single query.
       def update(id = :all, attributes)
         if id.is_a?(Array)
-          id.map { |one_id| find(one_id) }.each_with_index { |object, idx|
-            object.update(attributes[idx])
-          }
+          raise(ArgumentError, "the list of ids and the list of attributes do not have " \
+                "the same length.  id: #{id.size} attributes: #{attributes.size}") if id.size != attributes.size
```

Comment:
Minor style suggestion here, to avoid having the condition way to the end of the second line.

```suggestion
          if id.size != attributes.size
            raise ArgumentError,
                  "the list of ids and the list of attributes do not have " \
                  "the same length. id: #{id.size} attributes: #{attributes.size}"
          end
```